packages/wrangler/src/pages.tsx

```diff
@@ -824,11 +831,130 @@ const createDeployment: CommandModule<
       })
       .epilogue(pagesBetaWarning);
   },
-  handler: async (args) => {
-    const { directory, project } = args;
-    let { branch, commitHash, commitMessage, commitDirty } = args;
+  handler: async ({
+    directory,
+    projectName,
+    branch,
+    commitHash,
+    commitMessage,
+    commitDirty,
+  }) => {
+    const config = getConfigCache<PagesConfigCache>(
+      PAGES_CONFIG_CACHE_FILENAME
+    );
+    const isInteractive = process.stdin.isTTY;
+    const accountId = await requireAuth(config, isInteractive);
+
+    projectName ??= config.project_name;
+
+    if (!projectName && isInteractive) {
+      const existingOrNew = await new Promise<"new" | "existing">((resolve) => {
+        const { unmount } = render(
+          <>
+            <Text>
+              No project selected. Would you like to create one or use an
+              existing project?
+            </Text>
+            <SelectInput
+              items={[
+                {
+                  key: "new",
+                  label: "Create a new project",
+                  value: "new",
+                },
+                {
+                  key: "existing",
+                  label: "Use an existing project",
+                  value: "existing",
+                },
+              ]}
+              onSelect={async (selected) => {
+                resolve(selected.value as "new" | "existing");
+                unmount();
```

Comment:
Maybe it isn't an issue but calling unmount in an event handler feels funky. Perhaps we should call it after the resolve?

Reply:
This is how all the other dialogs are set up, and it seems to work well enough for them! https://github.com/cloudflare/wrangler2/blob/main/packages/wrangler/src/dialogs.tsx

Reply:
Oh, never mind then.